src/components/Filters/Filters.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { useState, useRef } from 'react';
+import styles from './Filters.module.css';
+import Button from '../Button';
+import { CheckboxField, FiltersDrawer } from '../FiltersDrawer/FiltersDrawer';
+import Icon from '../Icon';
+
+export type Props = {
+  /**
+   * Checkbox fields that will be displayed in the filters drawer.
+   */
+  checkboxFields: CheckboxField[];
+  /**
+   * CSS class names that can be appended to the component.
+   */
+  className?: string;
+  /**
+   * Callback called when filters drawer is closed.
+   */
+  onClose: (checkedValues: { [key: string]: boolean }) => void;
+};
+
+/**
+ * BETA: This component is still a work in progress and is subject to change.
+ *
+ * ```ts
+ * import {Filters} from "@chanzuckerberg/eds";
+ * ```'
+ *
+ * A filter component with a button that triggers a drawer of checkbox filters to be selected.
+ */
+export const Filters = ({ checkboxFields, className, onClose }: Props) => {
+  /**
+   * Maps the checkbox values to false initially to indicate that there are no filters selected.
+   * TODO: customize to allow initial filter values.
+   */
+  const checkedMap = {};
+  checkboxFields.forEach(({ checkboxes }) => {
+    checkboxes.forEach(({ value }) => {
+      checkedMap[value] = false;
+    });
+  });
+
+  /**
+   * Manages state of the applied checkboxes in the context of the filters component.
+   * It is only updated when the filters drawer is closed.
+   */
+  const [appliedCheckedBoxes, setAppliedCheckedBoxes] = useState({
+    ...checkedMap,
+  });
+
+  /**
+   * Manages the active state of the filters drawer.
+   */
+  const [isActive, setIsActive] = useState(false);
+
+  function closeFiltersDrawer(checkedValues: { [key: string]: boolean }) {
+    setTimeout(() => {
+      filtersButton?.current?.focus();
+    }, 1);
```

Comment:
what's the significance of waiting 1 ms?

Reply:
Good question, removed the `setTimeout`, and works, thanks

Reply:
another question mark from BM lol ... we may want to move it to the end of this function so focus is returned after `setIsActive` runs?

Reply:
I think the intention of the `setTimeout` was to have the `focus()` occur after `setIsActive` since `setState` is async. Since it's async, don't think moving the `focus()` to after `setIsActive` would change any functionality right now

Reply:
ahhh ok that makes sense! so should we keep the setTimeout? 🤔  and maybe leave that comment

Reply:
I don't think it's necessary to keep the `setTimeout` unless you think there is going to be some some crazy async stuff in the `onClose` that has to run before the focus occurs (such as a navigate away? or data fetching?)

Reply:
In my head it makes sense that the order of events should be close the drawer -> then return focus to the Filters button, and seems like removing the `setTimeout` would make it the reverse? It's up to you though (and lmk if I'm wrong)

Reply:
Yea that makes sense to me, will keep the `setTimeout`